homeassistant/components/nut/sensor.py

```diff
@@ -241,38 +243,43 @@ def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit
 
+    @property
+    def available(self):
+        """Return if the device is polling successfully."""
+        return self._available
+
     @property
     def device_state_attributes(self):
         """Return the sensor attributes."""
-        attr = dict()
-        attr[ATTR_STATE] = self.display_state()
-        return attr
+        return {ATTR_STATE: self._display_state}
 
-    def display_state(self):
+    def _format_display_state(self, status):
         """Return UPS display state."""
-        if self._data.status is None:
+        if status is None:
             return STATE_TYPES["OFF"]
         try:
-            return " ".join(
-                STATE_TYPES[state] for state in self._data.status[KEY_STATUS].split()
-            )
+            return " ".join(STATE_TYPES[state] for state in status[KEY_STATUS].split())
         except KeyError:
             return STATE_UNKNOWN
 
     def update(self):
         """Get the latest status and use it to update our sensor state."""
-        if self._data.status is None:
-            self._state = None
+        status = self._data.status
```

Comment:
I/O in properties is treacherous. The data update coordinator would be good here.

Reply:
Hard to track down for sure.  @mezz64 it looks like you added this in 6dfae7a259212222c3bd1703c95df2d77ac08664.   Would you consider converting it to `DataUpdateCoordinator`? https://developers.home-assistant.io/docs/integration_fetching_data/

Reply:
I'd love to, but i'm short on time these days and don't know when I'd be able to get to it.

Reply:
I'll add it to my queue :)

I'm going to do a config flow first

Reply:
https://github.com/home-assistant/core/pull/33457